code/modules/projectiles/ammo_datums.dm

```diff
@@ -190,6 +190,31 @@ GLOBAL_LIST_INIT(no_sticky_resin, typecacheof(list(/obj/item/clothing/mask/faceh
 			new_angle -= 360
 		new_proj.fire_at(shooter.Adjacent(target) ? target : null, main_proj.loc, source, range, speed, new_angle, TRUE) //Angle-based fire. No target.
 
+/// A variant of Fire_bonus_projectiles without fixed scatter and no link between gun and bonus_projectile accuracy
+/datum/ammo/proc/fire_directionalburst(obj/projectile/main_proj, atom/shooter, atom/source, range, speed, angle, target)
+	var/effect_icon = ""
+	var/proj_type = /obj/projectile
+	if(istype(main_proj, /obj/projectile/hitscan))
+		proj_type = /obj/projectile/hitscan
+		var/obj/projectile/hitscan/main_proj_hitscan = main_proj
+		effect_icon = main_proj_hitscan.effect_icon
+	for(var/i = 1 to bonus_projectiles_amount) //Want to run this for the number of bonus projectiles.
+		var/obj/projectile/new_proj = new proj_type(main_proj.loc, effect_icon)
+		if(bonus_projectiles_type)
+			new_proj.generate_bullet(bonus_projectiles_type)
+			var/obj/item/weapon/gun/g = source
+			if(source) //Check for the source so we don't runtime if we have bonus projectiles from a non-gun source like a Spitter
+				new_proj.damage *= g.damage_mult //Bonus or reduced damage based on damage modifiers on the gun.
+		else //If no bonus type is defined then the extra projectiles are the same as the main one.
+			new_proj.generate_bullet(src)
+
+		///Scatter here is how many degrees extra stuff deviate from the main projectile's firing angle. Fully randomised with no 45 degree cap like normal bullets
+		var/new_angle = angle + ( (rand(0, main_proj.ammo.bonus_projectiles_scatter)) * (prob(50) ? 1 : -1) )
+		if(new_angle < 0)
+			new_angle += 360
+		else if(new_angle > 360)
```

Comment:
dont need the else if since new_angle cant both be less than 0 and greater than 360

Reply:
making into a simple else.